src/BenchmarkDotNet/Running/ConsoleTitler.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+using System.IO;
+using BenchmarkDotNet.Portability;
+
+namespace BenchmarkDotNet.Running
+{
+    /// <summary>
+    /// Updates Console.Title, subject to platform capabilities and Console availability.
+    /// Restores the original (or fallback) title upon disposal.
+    /// </summary>
+    internal class ConsoleTitler : IDisposable
+    {
+        /// <summary>
+        /// Whether this instance has any effect. This will be false if the platform doesn't support Console retitling,
+        /// or if Console output is redirected.
+        /// </summary>
+        public bool IsEnabled { get; private set; }
+
+        private string oldConsoleTitle;
+
+        /// <summary>
+        /// Constructs a ConsoleTitler
+        /// </summary>
+        /// <param name="fallbackTitle">What to restore Console.Title to upon disposal (for platforms with write-only Console.Title)</param>
+        public ConsoleTitler(string fallbackTitle)
+        {
+            // Return without enabling if Console output is redirected, or if we're not on a platform that supports Console retitling.
+            if (Console.IsOutputRedirected || !PlatformSupportsTitleWrite())
+            {
+                return;
+            }
+
+            try
+            {
+                oldConsoleTitle = PlatformSupportsTitleRead() ? Console.Title : fallbackTitle;
+                IsEnabled = true;
+            }
+            catch (IOException)
```

Comment:
This will require a #pragma to suppress CA1416 - you OK with that?

Console.Title's get accessor is defined with [SupportedOSPlatform("windows")]

Reply:
Well, I suppose it makes sense to have the platform check for the get since that's currently the only supported platform. It can be updated in the future if that ever changes. But no platform check for the set. I still think the simpler try/catch makes sense, though, even with the platform check.